lib/libutils/isoc/bget_malloc.c

```diff
@@ -674,7 +681,14 @@ void *mdbg_malloc(const char *fname, int lineno, size_t size)
 {
 	struct mdbg_hdr *hdr;
 
-	COMPILE_TIME_ASSERT(sizeof(struct mdbg_hdr) == sizeof(uint32_t) * 4);
+	/*
+	 * Check struct mdbg_hdr doesn't get bad alignment.
+	 * This is required by C standard: the buffer returned from
+	 * malloc() should be aligned with a fundamental alignment.
+	 * For ARM32, the required alignment is 8. For ARM64, it is 16.
+	 */
+	COMPILE_TIME_ASSERT(
+		(sizeof(struct mdbg_hdr) % (__alignof(uintptr_t) * 2)) == 0);
```

Comment:
I don't understand this. How does it ensures that `raw_malloc()` returns a properly aligned pointer? (that is what we want, right?)